pkgs/development/interpreters/evcxr/withPackages.nix

```diff
@@ -0,0 +1,109 @@
+{ callPackage
+, fetchFromGitHub
+, lib
+, runCommand
+
+, python3
+
+, cargo
+, rustPlatform
+}:
+
+let
+  cratesIndex = fetchFromGitHub {
+    owner = "rust-lang";
+    repo = "crates.io-index";
+    rev = "d3be09973f4126cda5f8e8e77402e93be5f11bfe";
+    sha256 = "1pnsk6a7w71mv792rgf6l5jn3di0zmdibv8h4vbh0jwjr2w37yf2";
+  };
+
+  allPackageNames = runCommand "rust-package-names.nix" {} ''
+    echo "[" >> $out
+
+    find "${cratesIndex}" -type f -exec basename {} \; \
+      | awk '{ print "\""$0"\""}' \
+      | sort \
+      >> $out
+
+    echo "]" >> $out
+  '';
+
+  cargoToml = packageNames: runCommand "Cargo.toml" { buildInputs = [cargo]; } ''
+    # Generate Cargo.toml
+    cat <<EOT >> Cargo.toml
+    [package]
+    name = "rust_environment"
+    version = "0.1.0"
+    edition = "2018"
+
+    [dependencies]
+    EOT
+
+    for name in ${toString packageNames}; do
+      echo "$name = \"*\"" >> Cargo.toml
+    done
+
+    cp Cargo.toml $out
+  '';
+
+  cargoNix = packageNames: runCommand "Cargo-nix" { buildInputs = [cargo]; } ''
+    cp "${cargoToml packageNames}" Cargo.toml
+
+    # Generate lib.rs
+    mkdir -p "src"
+    touch src/lib.rs
+
+    # Set up fake index
+    mkdir -p index_base
+    ln -s ${cratesIndex} index_base/index
+
+    # Set up HOME with cargo config
+    mkdir -p home/.cargo
+    export HOME=$(pwd)/home
```

Comment:
Nit: Perhaps we should add quotes around the RHS.

Reply:
Done